crates/astria-sequencer/src/mempool/transactions_container.rs

```diff
@@ -434,7 +447,7 @@ pub(super) trait TransactionsForAccount: Default {
         ttx: TimemarkedTransaction,
         current_account_nonce: u32,
         current_account_balances: &HashMap<IbcPrefixed, u128>,
-    ) -> Result<(), InsertionError> {
+    ) -> Result<Option<[u8; 32]>, InsertionError> {
```

Comment:
This has a bad code smell. The behavior of the trait method can no longer be understood from by just looking at the signature, but I also need to understand the implementation of `nonce_replacement_enabled` to understand its behavior.

Now that the behavior of the two queues is so fundamentally different, they should be represented by two different types.